cmd/config/internal/commands/cmdlistsetters.go

```diff
@@ -63,25 +65,45 @@ func (r *ListSettersRunner) preRunE(c *cobra.Command, args []string) error {
 
 func (r *ListSettersRunner) runE(c *cobra.Command, args []string) error {
 	if setterVersion == "v2" {
-		if err := r.ListSetters(c, args); err != nil {
+		openAPIFileName, err := ext.OpenAPIFileName()
+		if err != nil {
 			return err
 		}
-		if r.IncludeSubst {
-			return r.ListSubstitutions(c, args)
+		r.List.OpenAPIFileName = openAPIFileName
+
+		openAPIPaths, err := pathutil.SubDirsWithFile(args[0], openAPIFileName)
+		if err != nil {
+			return err
+		}
+		if len(openAPIPaths) == 0 {
+			return errors.Errorf("unable to find %s in %s", openAPIFileName, args[0])
+		}
+
+		// list setters for all the subpackages with openAPI file paths
+		for _, openAPIPath := range openAPIPaths {
+			resourcePath := strings.TrimSuffix(openAPIPath, openAPIFileName)
+			fmt.Fprintf(c.OutOrStdout(), "%s\n", resourcePath)
+			if err := r.ListSetters(c, openAPIPath, resourcePath); err != nil {
+				return err
+			}
+			if r.IncludeSubst {
+				if err := r.ListSubstitutions(c, openAPIPath); err != nil {
+					return err
+				}
+			}
+			// clear setters and substitutions to hold next list them for next package
+			r.List.Setters = r.List.Setters[:0]
```

Comment:
This seems weird. Ideally we should refactor this so we don't have to clear slices inside other objects here. But at least we should just create new slices here instead of emptying existing ones.

Reply:
Done.